pkg/operator/ceph/object/controller_test.go

```diff
@@ -681,48 +659,79 @@ func TestCephObjectStoreControllerMultisite(t *testing.T) {
+		objectStore := &cephv1.CephObjectStore{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      objectStoreName,
+				Namespace: namespace,
+			},
+			TypeMeta: metav1.TypeMeta{
+				Kind: "CephObjectStore",
+			},
+			Spec: cephv1.ObjectStoreSpec{},
+		}
 
-	// Create a fake client to mock API calls.
-	cl := fake.NewClientBuilder().WithScheme(s).WithRuntimeObjects(object...).Build()
+		objectStore.Spec.Zone.Name = zoneName
+		objectStore.Spec.Gateway.Port = 80
+		if objectStoreName == customEndpointStore {
+			objectStore.Spec.Zone.Endpoint = customEndpoint
 
-	r := &ReconcileCephObjectStore{
-		client:              cl,
-		scheme:              s,
-		context:             c,
-		objectStoreContexts: make(map[string]*objectStoreHealth),
-		recorder:            record.NewFakeRecorder(5),
-		opManagerContext:    ctx,
-	}
+		}
 
-	_, err := r.context.Clientset.CoreV1().Secrets(namespace).Create(ctx, secret, metav1.CreateOptions{})
-	assert.NoError(t, err)
+		objects := []runtime.Object{
+			objectZone,
+			objectZoneGroup,
+			objectRealm,
+			objectStore,
+			cephCluster,
+		}
 
-	req := reconcile.Request{
-		NamespacedName: types.NamespacedName{
-			Name:      store,
-			Namespace: namespace,
-		},
-	}
+		// Create a fake client to mock API calls.
+		cl := fake.NewClientBuilder().WithScheme(s).WithRuntimeObjects(objects...).Build()
+
+		r := &ReconcileCephObjectStore{
+			client:              cl,
+			scheme:              s,
+			context:             c,
+			objectStoreContexts: make(map[string]*objectStoreHealth),
+			recorder:            record.NewFakeRecorder(5),
+			opManagerContext:    ctx,
+		}
+
+		_, err := r.context.Clientset.CoreV1().Secrets(namespace).Create(ctx, secret, metav1.CreateOptions{})
+		if !k8serrors.IsAlreadyExists(err) {
+			_, err = r.context.Clientset.CoreV1().Secrets(namespace).Update(ctx, secret, metav1.UpdateOptions{})
+			assert.NoError(t, err)
+		}
 
+		return r, objectStore, reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Name:      objectStoreName,
+				Namespace: namespace,
+			},
+		}
+
+	}
 	currentAndDesiredCephVersion = func(ctx context.Context, rookImage string, namespace string, jobName string, ownerInfo *k8sutil.OwnerInfo, context *clusterd.Context, cephClusterSpec *cephv1.ClusterSpec, clusterInfo *client.ClusterInfo) (*cephver.CephVersion, *cephver.CephVersion, error) {
 		return &cephver.Pacific, &cephver.Pacific, nil
 	}
 
+	r, objectStore, req := setupEnviroment(store)
 	t.Run("create an object store", func(t *testing.T) {
 		res, err := r.Reconcile(ctx, req)
 		assert.NoError(t, err)
 		assert.False(t, res.Requeue)
 		assert.True(t, calledCommitConfigChanges)
 		err = r.client.Get(ctx, req.NamespacedName, objectStore)
 		assert.NoError(t, err)
+		assert.Equal(t, objectStore.Status.Info["endpoint"], BuildDNSEndpoint(BuildDomainName(objectStore.Name, objectStore.Namespace), objectStore.Spec.Gateway.Port, false))
 	})
 
 	t.Run("delete the same store", func(t *testing.T) {
```

Comment:
Good unit test hygiene is that each test should be independent. `setupEnvironment()` should be called from inside of `t.Run()` to ensure that tests don't inadvertently share the same state and don't have inadvertent dependencies.

To keep the tests independent and still test deletion explicitly and separately, you can nest the tests together inside of a wrapping test.

```go
t.Run("normal object store", func(t *testing.T) {
    r, objectStore, req := setupEnviroment(store)

    t.Run("create object store", func(t *testing.T) {
        // ...
    }
    t.Run("delete object store", func(t *testing.T) {
        // ...
    }
}

t.Run("object store with custom zone endpoint", func(t *testing.T) {
    r, objectStore, req := setupEnviroment(customStore)

    t.Run("create object store", func(t *testing.T) {
        // ...
    }
    t.Run("delete object store", func(t *testing.T) {
        // ...
    }
}
```